src/app/core/admin/audit/audit-view-change.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component } from '@angular/core';
+
+import _isArray from 'lodash/isArray';
+import _isObject from 'lodash/isObject';
+import _keys from 'lodash/keys';
+
+import { BsModalRef } from 'ngx-bootstrap';
+
+@Component({
+	templateUrl: 'audit-view-details.component.html'
+})
+export class AuditViewDetailModal {
+
+	auditEntry: any;
+
+	constructor(
+		public modalRef: BsModalRef
+	) {}
+}
+
+@Component({
+	templateUrl: 'audit-view-change.component.html'
+})
+export class AuditViewChangeModal extends AuditViewDetailModal {
+
+	constructor(
+		public modalRef: BsModalRef
+	) {
+		super(modalRef);
+	}
+
+	// Derived from http://stackoverflow.com/a/1359808 and http://stackoverflow.com/a/23124958
+	sortObjectKeys(obj: any): any {
```

Comment:
This looks like a good candidate to in a util class somewhere that's injectable and sharable across the app.

Reply:
@tzfx Agreed! I think it makes sense to be its own pipe, since it's applicable to the templates. Will change.